### DISCUSSION TITLE
where is stored data form the tags marked as "registration enabled"

### DISCUSSION THREAD

**Discussion by mariorl** [frangoteam/FUXA#593] [Q&A]
I revised several .db files with sqlite browser within the project folder but for now did't found anything

**Comment by mariorl** [comment]
Let me explain the meaning of being able to access the tag value storage files. I have a Python/flask script who, using the bokeh library, takes the different values ​​that a porcess PLC variable had within certain time range, and assigns a color to each value. This is very useful because in the PLC programming part we usually assign values ​​to the states of a given process, for example 0=rest, 10=start, 20=production, etc. In this way, my script plot a Gantt-type chart in which we can see very clearly through which states our process passes, and how much time it invests in each one. But I can't find the file where Fuxa saves the values ​​and their timestamps. I assume that when I check to ON the "Registration enabled" button on a certain Fuxa/Device/Tag, is for datalogging it anywhere.
![imagen](https://user-images.githubusercontent.com/6344456/203134159-f8a19ad0-da62-4b2d-8755-3af932b10d95.png)

**Reply by unocelli** [reply]
Hi,
yes, you need to enable the registration for all tags whose history you want to have (internal device don't work).
<img width="496" alt="image" src="https://user-images.githubusercontent.com/48684550/203153666-ae30a23a-dc24-4017-ba38-2ef3feecf3da.png">

In `server/_db` folder you find the sqlite files (if you have enable sqlite as storage db)

**Reply by mariorl** [reply]
Thanks i found the values indeed at server/_db/archive/

**Reply by mariorl** [reply]
I noted, for example, an int var in Fuxa server is stored OK in _db. It is easy to test creating a input field in a view and modifiying its value (save value if checked is enabled). But a variable created in S7 is not stored, for example, I create an always increasing integer in my PLC, then declare the var in Fuxa, so that its increasing value is inmediately showed in device table, then I set to onm the propertyes for storage to 10 seconds and on every change, Restart the whole system, the S7 var is not stored.   (Note: I use NettoPlcsim who is very useful for debugging HMI apps without having a physical Siemens PLC)

**Reply by unocelli** [reply]
I have just tested and cannot confirm your issue. I suggest you recheck your settings.

**Reply by mariorl** [reply]
Oh, may be some wrong by my side, that's the reason for non openning an issue. The thing I noted is there is a _map database created in _db folder, and a correspondig  _0_  daq database db file. So that the _0_ database contains the values stored for the Fuxa server integer test variable I've created, and _map db file contains name and type of stored var. But both in the _0_ and _map db files there are nothing about the other S7 integer variables who I marked as storage. There are several .db files also in the _db folder, but these files are empty.
![FUXA](https://user-images.githubusercontent.com/6344456/204087799-fecebd2d-b7b9-49e5-980a-d2777824f624.png)

**Reply by unocelli** [reply]
Pro device there are 2 files:
- daq-map_[device ID].db contains the tags property 
![image](https://user-images.githubusercontent.com/48684550/204155571-4b27ab71-0f4d-4d1f-b998-67840bf5df81.png)

- daq-data_[device ID]_[first timestamp].db contains the tags value
![image](https://user-images.githubusercontent.com/48684550/204155743-e6947eba-27e3-4715-98da-902a602676c2.png)

you find the device ID by export it in json format

**Reply by mariorl** [reply]
Thanks for you fast reply. I just tested the issue with a variable storage of an actual S7 PLC, and storage just works. So the issue is concerning to the fact I used a simulated PLC in the previous tests. It's curious, because despite the var declared in Fuxa device/Tags table, is form a simulated PLC (Siemens PLCsim via NettoPlcSim), it appears in that table without errors and sowing its current value, but storage wont work for any reason. I'll do test with other ways of connecting to PLCsim in order to see if I can get it working and report here.
